File: python/approute/resources/services.py
from approute.models import Product, ProductListResponse, ProductStockResponse
from approute.resources._base import BaseResource


class ServicesResource(BaseResource):

    def list(self) -> ProductListResponse:
        """List all products/services in the catalog."""
        data = self._t.request("GET", "/services")
        return ProductListResponse.model_validate(data)

    def get(self, product_id: str) -> Product:
        """Get a single product/service by ID."""
        data = self._t.request("GET", f"/services/{product_id}")
        return Product.model_validate(data)

    def stock(self, product_id: str) -> ProductStockResponse:
        """Get stock info for a product."""
        data = self._t.request("GET", f"/services/{product_id}/stock")
        return ProductStockResponse.model_validate(data)

File: javascript/src/resources/services.ts
import { BaseResource } from "./base-resource.js";
import type {
  Product,
  ProductListResponse,
  ProductStockResponse,
} from "../models/index.js";

/**
 * Resource for the /services endpoints (product catalog).
 */
export class ServicesResource extends BaseResource {
  /**
   * List all products/services in the catalog.
   */
  async list(): Promise<ProductListResponse> {
    return this.transport.request<ProductListResponse>("GET", "/services");
  }

  /**
   * Get a single product/service by ID.
   */
  async get(productId: string): Promise<Product> {
    return this.transport.request<Product>("GET", `/services/${productId}`);
  }

  /**
   * Get stock info for a product.
   */
  async stock(productId: string): Promise<ProductStockResponse> {
    return this.transport.request<ProductStockResponse>(
      "GET",
      `/services/${productId}/stock`,
    );
  }
}
